### ISSUE TITLE
Razer Raiju not detected (I'm willing to help)

### ISSUE THREAD

**Issue by InvictusApollo** [Jays2Kings/DS4Windows#300]
Unfortunately Razer Raiju is not detected as DS4 and thus there is no touchpad functionality. I'm willing to help in anyway I can to implement support for this controller. I've sent an e-mail to jays@outlook.com on 22 May 2017. Please contact me through the e-mail I've sent it from if I can help. Title: "I wish to help with DS4Windows Razer Raiju support".

**Comment by danemadsen** [comment]
hello im kind of a noob to coding but i would like to see this implimented so by looking at some of the pull request i have gathered that we needed to add the pid to DS4Devices.cs

**Comment by danemadsen** [comment]
unsure as to what else needs to be done but device isnt recognized still

**Comment by InvictusApollo** [comment]
Have you tried this pids and vids?
VID_1532&PID_1000
VID_054C&PID_09CC
If I remember right, one of those sets are from my Raiju.
If you have everything set up, maybe try adding all the possible vids and pids and output log with device names. That way we will know if it is just a matter of adding Raiju id numbers. This way program will detect all system devices, so probably a try{}catch{} block should be added around some vital code.

**Comment by danemadsen** [comment]
Yep will do i already added 0x1000 as it is my raijus pid i wouldnt know where vid and device names need to go to impliment them

**Comment by danemadsen** [comment]
ok so did some more investigating and it seems i had to change the vid from the official playstation vid to razers own vid meaning i have made a verson of ds4 windows that is only compatible with the razer raiju. i tryed to implement both vid's but im sorry to say is to far above my skill in c++. so i will upload the source of what ive made it will work fine but no official playstation support

**Comment by InvictusApollo** [comment]
Great work danemadsen! You are awesome! Could you please also upload fully built version (exe file) for those who don't have necessary IDE or skills to build executable program?

**Comment by danemadsen** [comment]
sure thing not sure if github will let me upload compiled code but if not i will make a mediafire link

**Comment by danemadsen** [comment]
https://www.mediafire.com/?9ofjes1e50th7fi

**Comment by danemadsen** [comment]
also make sure you dont have razers official raiju driver installed and disable ps4 controller in steam

**Comment by MartelRagnarok** [comment]
Hmmm. Any idea how to permanently uninstall the official Razer Raiju driver?  I uninstall it, then I can see the controller in DS4Windows.. then seconds later windows10 re-installs the damn thing and the controller is gone again.

**Comment by InvictusApollo** [comment]
I honestly recommend uninstalling that spyware you use, called Windows 10 and installing Windows 7 :)
Other than that I think turning auto updates off would be enough.

**Comment by MartelRagnarok** [comment]
Sadly, that's not the case. I specifically disabled automatic driver installation... and it still does it. I even tried on a PC that didn't have the Razer drivers installed, and I still go the same issue. So it doesn't appear to be caused by the manually installed razer raiju driver. (That being said, I have no idea exactly what's being installed automatically.)

And finally. I'm not changing OSes over this. But I will note that I did test on windows 7, and was able to get DS4Windows working without issue. So that's valid advice.

**Comment by domdom90** [comment]
> also make sure you dont have razers official raiju driver installed and disable ps4 controller in steam

danemadsen or everyone, thank for the help, but when i connect the razer raiju is always on controller 2, the controller 1 is always connect to BT controller that i dont know about the controller, how to fix the problem? and are you get the rumble when you connect razer raiju with ds4? hopely you guys can help me, thank

**Comment by mika-n** [comment]
@domdom90  You may have better luck with Ryochan7 version of DS4Windows unless you really want to use this Jays2Kings V1.4.x version (which is not really developed anymore).
#573

**Comment by domdom90** [comment]
@mika-n you have many thank from me bro, the razer raiju working well in Ryochan7 version :D
